iceoryx_utils/test/moduletests/test_cxx_string.cpp

```diff
@@ -905,3 +1117,972 @@ TYPED_TEST(stringTyped_test, StringWithContentIsNotEmtpy)
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find(testStdString, 0);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find(testStdString, 50);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+/// @note template <typename T>
+/// iox::cxx::optional<uint64_t> find_first_of(const T& t, uint64_t pos = 0) const noexcept
+TYPED_TEST(stringTyped_test, FindFirstOfFailsForEmptyStringInEmptyString)
+{
+    using myString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = myString().capacity();
+    string<STRINGCAP + 5> testString;
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = this->testSubject.find_first_of("");
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString;
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TYPED_TEST(stringTyped_test, FindFirstOfForStringInEmptyStringFails)
+{
+    using myString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = myString().capacity();
+    string<STRINGCAP + 5> testString("a");
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = this->testSubject.find_first_of("a");
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString = "a";
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TEST(String100, FindFirstOfForStringInNotEmptyStringWorks)
+{
+    string<10> testString("R2-D2");
+    string<100> substring1("2");
+    auto res = testString.find_first_of(substring1);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1));
+
+    res = testString.find_first_of(substring1, 1);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1));
+
+    res = testString.find_first_of(substring1, 2);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(4));
+
+    string<100> substring2("D3R");
+    res = testString.find_first_of(substring2);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(0));
+
+    res = testString.find_first_of(substring2, 1);
+    EXPECT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(3));
+}
+
+TEST(String100, FindFirstOfForNotIncludedStringFails)
+{
+    string<100> testString("Kernfusionsbaby");
+    string<100> substring("cdG");
+    auto res = testString.find_first_of(substring);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find_first_of(substring, 0);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    res = testString.find_first_of(substring, 50);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TEST(String100, FindFirstOfForStringLiteralInNotEmptyStringWorks)
+{
+    string<100> testString1("Mueslimaedchen");
+    auto res = testString1.find_first_of("lima");
```

Comment:
Maybe permute the chars searched for, e.g. "mali", the result should be unaffected.

Reply:
done